src/main/scala/com/advancedtelematic/campaigner/http/GroupUpdateResolver.scala

```diff
@@ -14,21 +14,20 @@ import scala.concurrent.{ExecutionContext, Future}
 class GroupUpdateResolver(deviceRegistry: DeviceRegistryClient, resolver: ResolverClient, resolverUri: Uri)
                          (implicit db: Database, ec: ExecutionContext) extends UpdateSupport  {
 
-  def groupUpdates(ns: Namespace, groupId: GroupId, maxRequests: Int = 50): Future[Seq[Update]] = {
+  def groupUpdates(ns: Namespace, groups: Set[GroupId], maxRequests: Int = 50): Future[Seq[Update]] = {
 
-    def fetchDevices(offset: Long, limit: Long, requests: Int = 1): Future[Seq[DeviceId]] = {
-      deviceRegistry.devicesInGroup(ns, groupId, offset, limit).flatMap { devicesInBatch =>
-        if (devicesInBatch.isEmpty)
-          FastFuture.successful(devicesInBatch)
-        else if (requests >= maxRequests)
-          FastFuture.failed(Errors.TooManyRequestsToRemote)
-        else
-          fetchDevices(offset + devicesInBatch.size.toLong, limit, requests + 1).map { devicesInBatch ++ _}
-      }
-    }
+    def fetchDevices(offset: Long, limit: Long, requests: Int = 1): Future[Set[DeviceId]] =
+      Future
+        .sequence(groups.map(deviceRegistry.devicesInGroup(ns, _, offset, limit)))
```

Comment:
I think we need something like
```scala
def fetchDevices(groupId: GroupId, offset: Long, limit: Long, requests: Int = 1): Future[Set[DeviceId]] = ???

for {
  devices <- Future.sequence(groups.flatMap(groupId => fetchDevices(groupId, 0, 50)))
 ...
```
The problem with your implementation is it tries to fetch devices from all groups until at least one if not fully fetched.